src/LiveComponent/assets/src/live_controller.ts

```diff
@@ -391,90 +373,109 @@ export default class extends Controller implements LiveController {
+        this.pendingActions = [];
+        this.isRerenderRequested = false;
+        // we're making a request NOW, so no need to make another one after debouncing
+        this.#clearRequestDebounceTimeout();
+
+        // check if any unsynced inputs are now "in sync": their value matches what's in the store
+        // if they ARE, then they are on longer "unsynced", which means that any
+        // potential new values from the server *should* now be respected and used
+        this.unsyncedInputs.allMappedFields().forEach((element, modelName) => {
+            if (getValueFromInput(element, this.valueStore) === this.valueStore.get(modelName)) {
+                this.unsyncedInputs.remove(modelName);
+            }
+        });
 
         const fetchOptions: RequestInit = {};
         fetchOptions.headers = {
             'Accept': 'application/vnd.live-component+html',
         };
 
-        if (action) {
-            this.isActionProcessing = true;
+        const updatedModels = this.includeUpdatedModelsValue ? this.valueStore.updatedModels : false;
+        this.valueStore.updatedModels = [];
+        if (actions.length === 0 && this._willDataFitInUrl(this.valueStore.asJson(), params)) {
+            params.set('data', this.valueStore.asJson());
+            if (updatedModels) {
+                updatedModels.forEach((model) => {
+                    params.append('updatedModels[]', model);
+                });
+            }
+            fetchOptions.method = 'GET';
+        } else {
+            fetchOptions.method = 'POST';
+            fetchOptions.headers['Content-Type'] = 'application/json';
+            // TODO: I just moved under data
+            const requestData: any = { data: this.valueStore.all() };
+            if (updatedModels) {
+                requestData.updatedModels = updatedModels;
+            }
 
-            url += `/${encodeURIComponent(action)}`;
+            if (actions.length > 0) {
+                // one or more ACTIONs
+                if (this.csrfValue) {
+                    fetchOptions.headers['X-CSRF-TOKEN'] = this.csrfValue;
+                }
 
-            if (this.csrfValue) {
-                fetchOptions.headers['X-CSRF-TOKEN'] = this.csrfValue;
-            }
-        }
+                if (actions.length === 1) {
+                    // simple, single action
+                    // TODO : I just moved this data, it was query params
+                    requestData.args = actions[0].args;
 
-        let dataAdded = false;
-        if (!action) {
-            const dataJson = this.valueStore.asJson();
-            if (this._willDataFitInUrl(dataJson, params)) {
-                params.set('data', dataJson);
-                fetchOptions.method = 'GET';
-                dataAdded = true;
+                    url += `/${encodeURIComponent(actions[0].name)}`;
+                } else {
+                    // TODO: support on the server
+                    url += '/batch';
```

Comment:
```suggestion
                    url += '/_batch';
```

To avoid conflicts with a custom action.

Reply:
yea I was thinking about that too 👍